src/Visualization/Python/GenerateXdmf.py

```diff
@@ -319,11 +338,12 @@ def generate_xdmf(
       use_tetrahedral_connectivity: Optional. Use "tetrahedral_connectivity".
         Default: False
     """
-    h5files = [(h5py.File(filename, "r"), filename) for filename in h5files]
+    h5file_names = h5files
 
     if not subfile_name:
         subfiles = available_subfiles(
-            (h5file for h5file, _ in h5files), extension=".vol"
+            (h5py.File(filename, "r") for filename in h5file_names),
```

Comment:
You can just pass the list of filenames to `available_subfiles`:

```py
subfiles = available_subfiles(h5file_names, extension=".vol")
```

This is safer because I don't know if files are automatically closed if you open them in an iterator.